lib/DefineOpenCLWorkItemBuiltinsPass.cpp

```diff
@@ -212,16 +263,36 @@ bool DefineOpenCLWorkItemBuiltinsPass::defineGlobalSizeBuiltin(Module &M) {
 bool DefineOpenCLWorkItemBuiltinsPass::defineGlobalOffsetBuiltin(Module &M) {
   Function *F = M.getFunction("_Z17get_global_offsetj");
 
-  // If the builtin was not used in the module, don't create it!
-  if (nullptr == F) {
+  // If global offset support is disabled and the builtin not used in the
+  // module, don't create it. Otherwise, always define it as it's used in global
+  // ID calculations.
+  if (!clspv::Option::GlobalOffset() && (nullptr == F)) {
```

Comment:
get_global_id only uses get_global_offset if `clspv::Option::GlobalOffset()` returns true. It seems like this condition shouldn't define the function unless the option is set.

Reply:
This is actually what the code does but reversing the logic might make this clearer.

Reply:
I've reworked the code a bit to avoid re-declaring `get_global_offset` when it's already used and reworded the comments a bit. Hopefully this makes things clearer.